dev/TeachingTip/TeachingTip.cpp

```diff
@@ -0,0 +1,1507 @@
+            TeachingTipTestHooks::NotifyEffectiveBleedingPlacementChanged(*this);
+        }
+    }
+}
+
+void TeachingTip::UpdateDynamicBleedingContentPlacementToBottom()
+{
+    if (BleedingImagePlacement() == winrt::TeachingTipBleedingImagePlacementMode::Auto)
+    {
+        winrt::VisualStateManager::GoToState(*this, L"BleedingContentBottom"sv, false);
+        if (m_currentBleedingEffectivePlacementMode != winrt::TeachingTipBleedingImagePlacementMode::Bottom)
+        {
+            m_currentBleedingEffectivePlacementMode = winrt::TeachingTipBleedingImagePlacementMode::Bottom;
+            TeachingTipTestHooks::NotifyEffectiveBleedingPlacementChanged(*this);
+        }
+    }
+}
+
+void TeachingTip::OnIsOpenChanged()
+{
+    if (IsOpen())
+    {
+        m_lastCloseReason = winrt::TeachingTipCloseReason::Programmatic;
+        if (m_target)
+        {
+            m_currentTargetBounds = m_target.get().TransformToVisual(nullptr).TransformBounds({
+                0.0,
+                0.0,
+                static_cast<float>(m_target.get().as<winrt::FrameworkElement>().ActualWidth()),
+                static_cast<float>(m_target.get().as<winrt::FrameworkElement>().ActualHeight())
+                });
+        }
+        if (!m_popup)
+        {
+            m_popup.set(winrt::Popup());
+            m_popup.get().Child(*this);
+            m_popupClosedRevoker = m_popup.get().Closed(winrt::auto_revoke, { this, &TeachingTip::OnPopupClosed });
+            if (IsLightDismissEnabled())
+            {
+                m_popup.get().IsLightDismissEnabled(true);
+            }
+        }
+        if (!m_contractAnimation)
+        {
+            CreateContractAnimation();
+        }        
+        if (!m_expandAnimation)
+        {
+            CreateExpandAnimation();
+        }
+
+        if (!m_popup.get().IsOpen())
+        {
+            // We are about to begin the process of trying to open the teaching tip, so notify that we are no longer idle.
+            if (m_isIdle)
+            {
+                m_isIdle = false;
+                TeachingTipTestHooks::NotifyIdleStatusChanged(*this);
+            }
+            m_popup.get().IsOpen(true);
+            if (m_beakOcclusionGrid)
+            {
+                StartExpandToOpen();
+            }
+            else
+            {
+                m_startAnimationInOnApplyTemplate = true;
+            }
+        }
+        else
+        {
+            // We have become Open but our popup was already open. This can happen when a close is canceled by the closing event, so make sure the idle status is correct.
+            if (!m_isIdle && !m_isExpandAnimationPlaying && !m_isContractAnimationPlaying)
+            {
+                m_isIdle = true;
+                TeachingTipTestHooks::NotifyIdleStatusChanged(*this);
+            }
+        }
+    }
+    else
+    {
+        if (m_popup)
```

Comment:
I think you will need to clear the Child of the Popup here. Otherwise you will have a permanent reference cycle between the TeachingTip and the Popup.